### PR TITLE
Remove `AcceleratorConnector.tpu_id`

### PR DESCRIPTION
**Pull request by DuYicong515**
## What does this PR do?
Part of https://github.com/PyTorchLightning/pytorch-lightning/issues/11449: deprecate unused properties in accelerator_connector: tpu_id

`AccleratorConnector.tpu_id` was not used internally, and was not meant to be public. Remove it in favor of using `Trainer.device_ids` to check the TPU id when use TPU accelerator directly. 

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/models/test_tpu.py
Comment (DuYicong515) [inline tests/models/test_tpu.py:245]: Remove this test since `AcceleratorConnector.tpu_id` is removed, we don't need to test if it's set as expected. Its replacement `Trainer.device_ids` on TPU accelerator is tested in `tests/accelerators/test_tpu.py:: test_trainer_config_device_ids `

https://github.com/PyTorchLightning/pytorch-lightning/blob/ed2bcc5ab3e8eab45df17fee23c24d2ad8a20687/tests/accelerators/test_tpu.py#L328

File: pytorch_lightning/trainer/connectors/accelerator_connector.py
Comment (krshrimali) [inline pytorch_lightning/trainer/connectors/accelerator_connector.py:811]: Should we also add a deprecation message? Like how it was done [here](https://github.com/PyTorchLightning/pytorch-lightning/pull/12262/files#diff-667a2513c158c2b01d73b479e4dea75587d8531f9cd286891d34570a2fd145dcR2056-R2062).

File: tests/models/test_tpu.py
Comment (krshrimali) [reply tests/models/test_tpu.py:245]: Thanks for this comment, @DuYicong515 - certainly helps for easier review. Is the alternate test you mentioned currently skipped? Do you think we should keep this test if we have a deprecated version of `tpu_id` until we remove the skip for the mentioned test below? (cc: @kaushikb11)

https://github.com/PyTorchLightning/pytorch-lightning/blob/523200971d96c6bb40262d0ca0f25c8b8ecea7b4/tests/accelerators/test_tpu.py#L316

**Review by krshrimali — COMMENTED** [review]
Thanks, @DuYicong515! Left a couple of comments, looking forward to your thoughts on them. :)

**Review by ananthsub — APPROVED** [review]

File: pytorch_lightning/trainer/connectors/accelerator_connector.py
Comment (DuYicong515) [reply pytorch_lightning/trainer/connectors/accelerator_connector.py:811]: This is being removed directly, instead of being deprecated. The acclerator_connector and its properties are not user facing, they are very PyTorchLightning internal.

File: tests/models/test_tpu.py
Comment (DuYicong515) [reply tests/models/test_tpu.py:245]: Thanks @krshrimali  for reviewing! 

The newly added tests passes [in a previous CI test](https://app.circleci.com/pipelines/github/PyTorchLightning/pytorch-lightning/54466/workflows/709f3bc4-0d55-48ef-9d64-f85be1a64936/jobs/152737),  however we have to disable it for now since it leads to timeout due to the use of `@pl_multiprocess_test`.  Enabling it would require optimizing existing tests and TPU CI process which requires more efforts and investigations, it's tracked in #12335.

I feel `tpu_id` property was already PyTorchLightning internal facing, and it was not used within the project, it's safe to just remove the relevant tests here. Feel free to share your thoughts here.

Comment (krshrimali) [reply tests/models/test_tpu.py:245]: Thanks for the context, @DuYicong515 - the reasoning you shared here, and in [another comment](https://github.com/PyTorchLightning/pytorch-lightning/pull/12387#discussion_r830756792) makes sense to me. Let's wait for the code-owners approval, but it sounds good to me. :)

Comment (krshrimali) [reply tests/models/test_tpu.py:245]: Just found this test which is using `tpu_id`: https://github.com/PyTorchLightning/pytorch-lightning/blob/523200971d96c6bb40262d0ca0f25c8b8ecea7b4/tests/models/test_tpu.py#L308-L316

Should this be updated as well?

File: pytorch_lightning/trainer/connectors/accelerator_connector.py
Comment (krshrimali) [reply pytorch_lightning/trainer/connectors/accelerator_connector.py:811]: Thanks, agreed, I didn't find any mention of this in the docs as well. Should be pytorch lightning internal, as you mentioned. Resolving this conversation. :)

File: tests/models/test_tpu.py
Comment (DuYicong515) [reply tests/models/test_tpu.py:245]: Yes, updating it now. Just noticing the TPU test failures as well. Thanks for the catch!

File: tests/models/test_tpu.py
Comment (DuYicong515) [inline tests/models/test_tpu.py:292]: Tests verifies `tpu_id` is removed, since this internal property is removed in this PR. 

The tpu_cores mis-configuration tests are merged with `test_tpu_misconfiguration()`

File: tests/models/test_tpu.py
Comment (krshrimali) [reply tests/models/test_tpu.py:245]: Awesome, thanks!

File: pytorch_lightning/trainer/connectors/accelerator_connector.py
Comment (kaushikb11) [reply pytorch_lightning/trainer/connectors/accelerator_connector.py:811]: We could remove it directly

**Review by kaushikb11 — APPROVED** [review]
Thanks for the clean up!

File: tests/models/test_tpu.py
Comment (rohitgr7) [reply tests/models/test_tpu.py:292]: what about the cases where error is not expected? are we testing them somewhere?

**Review by rohitgr7 — APPROVED** [review]

**Review by Borda — APPROVED** [review]

File: tests/models/test_tpu.py
Comment (DuYicong515) [reply tests/models/test_tpu.py:306]: The `device_ids` tests are in `tests/accelerators/test_tpu.py:: test_trainer_config_device_ids `

https://github.com/PyTorchLightning/pytorch-lightning/blob/ed2bcc5ab3e8eab45df17fee23c24d2ad8a20687/tests/accelerators/test_tpu.py#L328

More details are in my above comments.

**Review by tchaton — APPROVED** [review]
Nice work !

File: tests/models/test_tpu.py
Comment (rohitgr7) [reply tests/models/test_tpu.py:292]: then it's good to go 🚀

**Pull request merged**
